### PR TITLE
pass sectionId to RubricFloatingActionButton via redux

### PR DESCRIPTION
**Pull request by davidsbailey**
gets RubricFloatingActionButton connected to redux, and uses redux to send sectionId to that component. This is necessary to unblock other work in the case where section_id is not present in the url.

## Screenshots

Note that sectionId is undefined the first time the FAB is rendered, so any code which depends on sectionId will have to account for that.
Screenshot 2023-12-12 at 3 48 53 PM

## Testing story

* unit tests passing locally

## Future work

* pass the sectionId down to the child components to the place where it is needed (in RubricSettings)

### CODE REVIEW

**Comment by davidsbailey** [comment]
@cearachew here you go! 

@bethanyaconnor since you have familiarity with these components, can you chime in on whether you see any problems with this approach, to connect redux to the top component in the chain? my reason for doing this is so that we don't run into problems with unit tests, which is somewhat speculative -- IIRC it is hard to deep-render a component in a unit test, if one of its children is redux-connected.

**Comment by bethanyaconnor** [comment]
[full disclosure -- not a redux expert] I think this makes sense! It depends a little bit where this prop is needed -- if it's needed only on a component at the bottom of the chain, I think it could make sense there as well. But I generally like this component as being the entry point of the rubrics feature and if the rubrics feature needs something from redux, this is the entry point for that as well.

**Comment by davidsbailey** [comment]
thank you Bethany! @cearachew, if you want to be the reviewer, one idea is for me to merge this into staging (once it opens) and then you can merge in latest staging and depend on this in your PR.

**Comment by cearachew** [comment]
(disclaimer- also extremely not a redux expert) - I like this at the top of the rubrics chain- from working on AITT-365 and AITT-366 today, I think those will also need information from redux (the sets of students and sections), so imo it will be useful to have it here for other features too

File: apps/src/templates/rubrics/RubricFloatingActionButton.jsx
Comment (cearachew) [inline apps/src/templates/rubrics/RubricFloatingActionButton.jsx:102]: Just so I know for the future- this is also where I would add any other states that are needed from redux, correct? Along with adding them to the import on line 15?

**Review by cearachew — APPROVED** [review]
With a heavy disclaimer that I'm unfamiliar with redux, this looks ok to me so :shipit: and thank you!

File: apps/src/templates/rubrics/RubricFloatingActionButton.jsx
Comment (davidsbailey) [reply apps/src/templates/rubrics/RubricFloatingActionButton.jsx:102]: yes, this is where you would pull any other data that's in redux into any rubric-related components. if the thing you need is in teacherSectionsRedux.js then the data you need may already be in redux. depending on what you need, you might need to do extra steps to put the data into redux. 

In this case I am actually having trouble finding out where the data is being put into redux, but i convinced myself that I could rely on it being there because it is already being depended upon by the teacher panel.

Comment (davidsbailey) [reply apps/src/templates/rubrics/RubricFloatingActionButton.jsx:102]: Found it: https://github.com/code-dot-org/code-dot-org/blob/4fa2b9f0ebc62752f5cb13691b597c62172fc7e3/apps/src/code-studio/components/progress/teacherPanel/TeacherPanel.jsx#L104 this data is set in response to an ajax request once the page has already loaded, as opposed to being set at initial page load time.

**Pull request merged**
